crates/bitwarden-uniffi/src/log_callback.rs

```diff
@@ -0,0 +1,109 @@
+use std::sync::Arc;
+
+use tracing_subscriber::{Layer, layer::Context};
+/// Callback interface for receiving SDK log events
+/// Mobile implementations forward these to Flight Recorder
+#[uniffi::export(with_foreign)]
+pub trait LogCallback: Send + Sync {
+    /// Called when SDK emits a log entry
+    ///
+    /// # Parameters
+    /// - level: Log level ("TRACE", "DEBUG", "INFO", "WARN", "ERROR")
+    /// - target: Module that emitted log (e.g., "bitwarden_core::auth")
+    /// - message: The log message text
+    ///
+    /// # Returns
+    /// Result<(), BitwardenError> - mobile implementations should catch exceptions
+    /// and return errors rather than panicking
+    fn on_log(&self, level: String, target: String, message: String) -> crate::Result<()>;
+}
+
+/// Custom tracing Layer that forwards events to UNIFFI callback
+pub(crate) struct CallbackLayer {
+    callback: Arc<dyn LogCallback>,
+}
+impl CallbackLayer {
+    pub(crate) fn new(callback: Arc<dyn LogCallback>) -> Self {
+        Self { callback }
+    }
+}
+impl<S> Layer<S> for CallbackLayer
+where
+    S: tracing::Subscriber,
+{
+    fn on_event(&self, event: &tracing::Event<'_>, _ctx: Context<'_, S>) {
+        let metadata = event.metadata();
+        // CRITICAL: Filter out our own error messages to prevent infinite callback loop
+        if metadata.target() == "bitwarden_uniffi::log_callback" {
+            return; // Platform loggers still receive this for debugging
+        }
+        let level = metadata.level().to_string();
+        let target = metadata.target().to_string();
+        // Format event message
+        let mut visitor = MessageVisitor::default();
+        event.record(&mut visitor);
+        let message = visitor.message;
+        // Forward to UNIFFI callback with error handling
+        if let Err(e) = self.callback.on_log(level, target, message) {
+            // Error logged with explicit target for filtering above
+            // Platform loggers receive this for mobile team debugging
+            tracing::error!(target: "bitwarden_uniffi::log_callback", "Logging callback failed: {:?}", e);
+        }
+    }
+}
+/// Visitor to extract message from tracing event
+///
+/// **Why only record_debug is implemented:**
+///
+/// MessageVisitor implements only record_debug because it's the ONLY required method.
+/// The tracing::field::Visit trait provides default implementations for all other
+/// record methods (record_str, record_i64, record_u64, record_bool, etc.) that
+/// forward to record_debug. The SDK uses % and ? format specifiers which route
+/// through record_debug via tracing's default implementations.
```

Comment:
I don't understand what this comment is trying to communicate. Seems worded badly.